app/src/ui/lib/update-store.ts

```diff
@@ -160,10 +161,24 @@ class UpdateStore {
       return
     }
 
+    let updatesURL = __UPDATES_URL__
+
+    // If the app is running under Rosetta (i.e. it's a macOS x64 binary running
+    // on an arm64 machine), we need to tweak the update URL here to point at
+    // the arm64 binary.
+    if (
+      enableUpdateFromRosettaToARM64() &&
+      remote.app.runningUnderRosettaTranslation === true
+    ) {
+      const url = new URL(updatesURL)
+      url.searchParams.set('architecture', 'arm64')
+      updatesURL = url.toString()
+    }
```

Comment:
This is needed because `__UPDATES_URL__` is set at compile time, so we can't set it to look for arm64 for users who downloaded the x64 build to then run it on an Apple Silicon machine.